WebAppManifestClient.js

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import {httpClient} from '@digitalbazaar/http-client';
+import {getWebAppManifestIcon} from './manifest.js';
+
+const DEFAULT_HEADERS = {Accept: 'application/manifest+json, application/json'};
+
+export class WebAppManifestClient {
+  constructor({
+    defaultHeaders,
+    agent,
+    manifestProxyHost,
+    manifestProxyPath = '/manifest.json'
+  } = {}) {
+    this.defaultHeaders = {...DEFAULT_HEADERS, ...defaultHeaders};
+    this.agent = agent;
+    this.manifestProxyHost = manifestProxyHost;
+    this.manifestProxyPath = manifestProxyPath;
+  }
+
+  async getManifest({origin}) {
```

Comment:
Here and on all public methods:

```suggestion
  async getManifest({origin} = {}) {
```